### ISSUE TITLE
Review Weight and Distance

### ISSUE THREAD

**Issue by itadapter** [azist/azos#43]
why one of them holds value in unit, another does not?

**Comment by sergey-msu** [comment]
The code in Weight.cs is pretty strange. In .ctor one passes value which then transforms to ValueInGrams and the back to original value (which saves in Value). The only effect is corruption of original value due to division, rounding and multiplication operations.

**Comment by itadapter** [comment]
yes, he was trying to hold the value in a a uniform unit internally.
These two classes need to be re-written without looking at existing implementation but keeping the interface

**Comment by itadapter** [comment]
Re #512
